examples/header-group.html

```diff
@@ -0,0 +1,80 @@
+<template id=header-group>
+
+<style>
+
+  header-group
+    { display: block }
+
+  header-group
+    hgroup
+      { background: #bada55; color: var(--darkest-color) }
+
+</style>
+
+<hgroup>
+  <h1><a href=/examples/index.html>snuggsi ツ</a></h1>
+  <slot name=header><h2>try an example</h2></slot>
+</hgroup>
+
+</template>
+
+<script>
+const tag = `header-group`
+
+Element `header-group`
+
+(class extends HTMLElement {
+
+  static onimportload (event) {
+    let template = event.target.import
+      .querySelector ('template')
+
+      document
+        .querySelector('head')
+        .appendChild (template)
+
+      // possibly should be adopting node into document
+      // and should do it on DOMContentLoaded hook up onimportload
+
+    const shadow = function (element) {
+      let fragment = template.content.cloneNode (true)
+
+      fragment.slots =
+        Array.from (fragment.querySelectorAll ('slot'))
+
+      element.slots =
+        Array.from (element.querySelectorAll ('[slot]'))
+
+      element.slots.map (function (namedslot) {
+        fragment.slots
+          .filter (slot =>
+            (slot.getAttribute ('name') === namedslot.getAttribute ('slot'))
+          )
+          .map (slot => {
+            slot.innerHTML = ''
+            , slot.appendChild (namedslot)
+          })
+      })
+
+      element.innerHTML = ''
+      element.append (fragment)
+    }
+
+    Array.from
+      // should be using currentScript ?
+      (document.getElementsByTagName (tag))
+      .map (shadow)
+  }
+
+  initialize () {
+    const selector = `link#${tag}, link[href='${tag}.html']`
```

Comment:
Removing `.html` file extension assumption.

Theoretically following resources should work for dynamic scripts:
`<link rel=import href=//example.com/foo-bar>`
`<link rel=import href=//example.com/foo-bar.html>`
`<link rel=import href=//example.com/foo-bar.asp>`
`<link rel=import href=//example.com/foo-bar.foo?what=ever&baz=bat>`

/cc @mrbernnz @VicenteRD